lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyCompletionParticipant.java

```diff
@@ -72,6 +74,38 @@ public void onXMLContent(ICompletionRequest request, ICompletionResponse respons
                     existingFeatures, featureName, featureMgrNode);
             featureCompletionItems.stream().forEach(item -> response.addCompletionItem(item));
         }
+        if (parentElement.getTagName().equals(LibertyConstants.PLATFORM_ELEMENT)) {
+            buildPlatformCompletionItems(request, response, parentElement);
+        }
+    }
+
+    /**
+     * build completion items for platforms
+     * @param request request
+     * @param response response
+     * @param parentElement parent element xml dom
+     */
+    private static void buildPlatformCompletionItems(ICompletionRequest request, ICompletionResponse response, DOMElement parentElement) {
+        DOMNode platformTextNode = (DOMNode) parentElement.getChildNodes().item(0);
+        String platformName = platformTextNode != null ? platformTextNode.getTextContent() : null;
+        LibertyRuntime runtimeInfo = LibertyUtils.getLibertyRuntimeInfo(request.getXMLDocument());
+        String libertyVersion =  runtimeInfo == null ? null : runtimeInfo.getRuntimeVersion();
+        String libertyRuntime =  runtimeInfo == null ? null : runtimeInfo.getRuntimeType();
+
+        final int requestDelay = SettingsService.getInstance().getRequestDelay();
+        //get all platforms
+        Set<String> platforms = FeatureService.getInstance().getAllPlatforms(libertyVersion, libertyRuntime, requestDelay,
```

Comment:
Not sure why we would need the libertyVersion and libertyRuntime to get all the platforms. We will use the cached features.json for this right? Also, I don't think there is any logic here to remove platform values that are already specified on another `<platform>` element. It's better if we don't let them select a value that is already present, because they will just end up with a diagnostic stating they can have the duplicate value.

Reply:
Actually I see why the libertyVersion and libertyRuntime are passed now. Ignore that part of my comment.

Reply:
This was not resolved. So I added a comment to the new PR.